src/main/java/org/medicmobile/webapp/mobile/XWalkMigration.java

```diff
@@ -167,10 +172,21 @@ private File getStorageRootFromFiles(File filesDir) {
 	}
 
 	private void deleteRecursive(File fileOrDirectory) {
-		if (fileOrDirectory.isDirectory())
-			for (File child : fileOrDirectory.listFiles())
+		if (fileOrDirectory == null) {
+			return;
+		}
+
+		if (fileOrDirectory.isDirectory()) {
+			File[] files = fileOrDirectory.listFiles();
+			if (files != null) for (File child : files) {
 				deleteRecursive(child);
+			}
```

Comment:
This inline `for` is a little non-standard. How about...

```suggestion
			if (files != null) {
				for (File child : files) {
					deleteRecursive(child);
				}
			}
```

Reply:
Done